lib/Support/CMakeLists.txt

```diff
@@ -23,11 +23,21 @@ revng_add_library_internal(
   ProgramCounterHandler.cpp
   ResourceFinder.cpp
   SelfReferencingDbgAnnotationWriter.cpp
-  Statistics.cpp)
+  Statistics.cpp
+  GzipTarFile.cpp)
 
 llvm_map_components_to_libnames(LLVM_LIBRARIES Support Core Object)
 
-target_link_libraries(revngSupport ${LLVM_LIBRARIES})
+find_package(Boost REQUIRED COMPONENTS iostreams)
+include(FindLibArchive)
+
+if(NOT LibArchive_FOUND)
+  message(SEND_ERROR "libarchive not found")
+endif()
+
+target_include_directories(revngSupport PUBLIC ${Boost_INCLUDE_DIRS})
+target_link_libraries(revngSupport ${LibArchive_LIBRARIES} ${Boost_LIBRARIES}
+                      ${LLVM_LIBRARIES})
```

Comment:
> Iostreams is header only.

```
$ l root/lib/libboost_iostreams.so.1.81.0 
-rwxr-xr-x 1 user user 111K Aug  1 10:28 root/lib/libboost_iostreams.so.1.81.0
```

> Also `target_include_directories` should not be strictly required

I'll check if it's needed

Reply:
Ah shit. I googled this but I misread.

I really don't want to link against boost.
We're currently building boost with exceptions and RTTI and that does not play well with our way of building things.

I think we don't really need a new dependency just for gzip.
We have libarchive, let's use that one.

Sorry, but I misread. If I read correctly, I would have refused iostreams in the first place.

Reply:
> We can't use that one, we need to have finer-grain control

Like what?

Reply:
> Like what?

When writing the tar archive we need to:
* Write a compressed header
* Write the compressed file contents (in a separate gzip stream)
* Write the compressed trailing passing (also in a separate gzip stream)

libarchive does not offer this functionality

Reply:
In theory we could, but then we'd have to convert thing into `libarchive`-specific data structures which isn't great ([example](https://github.com/libarchive/libarchive/blob/dff77a6783e5de548ceb265aad1f7a74261be390/libarchive/archive_read_support_filter_gzip.c#L104)) (they do weird vtable stuff in c)

I'm looking at what libarchive does for streaming compression, which seems to be to use `deflateInit2` from zlib. I think we can use that instead of boost